src/javascript/app/pages/form/form-component.jsx

```diff
@@ -0,0 +1,421 @@
+            if (!endtime_data.show_datepicker) {
+                Array.from(element.options).map((option) => {
+                    if (moment(option.text).format('YYYY-MM-DD') === value) {
+                        option.setAttribute('selected', true);
+                        option.setAttribute('data-value', newDate);
+                    } else {
+                        option.setAttribute('selected', false);
+                        option.setAttribute('data-value', newDate);
+                    }
+                });
+            } else {
+                element.setAttribute('data-value', newDate);
+            }
+            eventDispatcher(element, 'change');
+        }
+    };
+
+    const updateFormField = (elementId, value, eventType) => {
+        const element = common_functions.getElementById(elementId);
+        if (elementId === 'callputequal') {
+            element.checked = !+is_equal;
+        } else {
+            element.value = value;
+        }
+        eventDispatcher(element, eventType);
+    };
+
+    const getMessage = (form) => {
+        if (form === 'highlowticks') {
+            return localize('This contract type only offers 5 ticks');
+        }
+        return null;
+    };
+
+    const handleAmountChange = (e, id) => {
+        updateFormField(id, e.target.value, 'input');
+    };
+
+    const findTextByValue = (arr, value) => arr.find(item => item.value === value)?.text || null;
+    
+    const isEmpty = (obj) => Object.keys(obj).length === 0;
+    if (isEmpty(tradeData)) {
+        return null;
+    }
+
+    const getMinMaxDate = () => {
+        const { minDate: min, maxDate: max } = endtime_data.datepicker_config;
+        const today = new Date();
+    
+        const minDate = new Date(today);
+        minDate.setDate(today.getDate() + min);
+    
+        const maxDate = new Date(today);
+        maxDate.setDate(today.getDate() + max);
+    
+        return { minDate, maxDate };
+    };
+
+    const payoutTypeOptions = [
+        { text: localize('Stake'), value: 'stake' },
+        { text: localize('Payout'), value: 'payout' },
+    ];
+
+    return (
+        <BreakpointProvider>
+            <div className='quill-form-container'>
+                {contractForms.includes(formName) && (
+                    <>
+                        {formName === 'highlowticks' && (
+                            <div className='section-msg-container'>
+                                <SectionMessage status='info' message={getMessage(formName)} />
+                            </div>
+                        )}
+                        {formName === 'reset' && reset_message && (
+                            <div className='section-msg-container'>
+                                <SectionMessage status='info' message={reset_message} />
+                            </div>
+                        )}
+                    
+                        <div className='quill-form-rows'>
+
```

Comment:
just a question: do we need extra line breaks in jsx here and below? :)

Reply:
nope. I just kept line between for easy readability for me :) will remove this.